### PR TITLE
feat: at import memory

### PR DESCRIPTION
**Pull request by mertmit**
## Change Summary

Re #4023 #4176
- Dynamic EntityMap (uses SQLite)
- Keep memory foot-print limited for import (only limited number of records loaded into memory, others are kept in temporary SQLite db)

## Change type

- [x] feat: (new feature for the user, not a new feature for build script)

### CODE REVIEW

File: packages/nocodb/src/lib/meta/api/sync/helpers/EntityMap.ts [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+  async init () {
+    if (!this.initialized) {
+      this.db = await this.db;
+      this.initialized = true;
+    }
+  }
+
+  destroy() {
+    if (this.initialized && this.db) {
+      this.db.close();
```
Comment (pranavxc) [inline packages/nocodb/src/lib/meta/api/sync/helpers/EntityMap.ts:35]: `this.db.destry()` or `this.db.close()`?

File: packages/nocodb/src/lib/meta/api/sync/helpers/EntityMap.ts [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    if (revertKey(key) !== key) {
+      res[revertKey(key)] = res[key];
+      delete res[key];
+    }
+  }
+  return res;
+}
+
+function processKey(key) {
+  return key.replace(/'/gi, "''").replace(/[A-Z]/g, (match) => `_${match}`);
```
Comment (pranavxc) [inline packages/nocodb/src/lib/meta/api/sync/helpers/EntityMap.ts:192]: `i` modifier is not required here `/'/gi` => `/'/g`

File: packages/nocodb/src/lib/meta/api/sync/helpers/EntityMap.ts [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+}
+
+function revertKey(key) {
+  return key.replace(/''/gi, "'").replace(/_[A-Z]/g, (match) => match[1]);
```
Comment (pranavxc) [inline packages/nocodb/src/lib/meta/api/sync/helpers/EntityMap.ts:196]: `i` modifier is not required here `/''/gi` => `/''/g`

**Review by pranavxc — APPROVED** [review]

**Review by dstala — APPROVED** [review]

**Comment by wingkwong** [comment]
Let's wait for the final tests. tag was missing in previous run.

**Pull request merged**